### PR TITLE
Filtering pytng reader warnings (resolves #3986)

### LINKED ISSUE #3986: pytng reader tests generate a lot of warnings
## Is your feature request related to a problem? ##

Running tests generates 611 warnings, by comparison PDB coordinate tests (which is generally the worst for the coordinate readers) only generates 268.

A lot of these warnings are repeated stride block issues and therefore could easily be filtered out.

## Describe the solution you'd like ##
Add some warnings filters to the TNG tests.

## Additional context ##
Issue #2980

### PR DESCRIPTION
**Pull request by ooprathamm**
Fixes #3986

Changes made in this Pull Request:
 -  added filter for repeated warning in test_tng.py

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/setup.cfg [hunk 1]
```diff
@@ -18,6 +18,8 @@ filterwarnings=
     ignore:Reader has no dt information, set to 1.0 ps
     # NamedStream warnings
     ignore:Constructed NamedStream:RuntimeWarning
+    # Stride block warnings 
+    ignore:Stride of block:UserWarning
```
Comment (IAlibay) [inline testsuite/setup.cfg]: We specifically do not want to do a block ignore for UserWarnings across our tests. The aim of the issue is to have specific filterwarnings on the pytng tests for warnings we specifically expect that each test should raise.

**Review by IAlibay — CHANGES_REQUESTED** [review]

**Comment by ooprathamm** [comment]
@IAlibay I hope this is what you meant. The filter has been applied specifically for test_tng.py

Current captured warnings -
image

**Comment by orbeckst** [comment]
@ooprathamm could you please edit your original PR comment at the top to **include the issue number** that you're fixing as `Fixes #3986` ? This links your PR to the issue and is an important part of organizing work on the project. Thanks.

**Comment by hmacdope** [comment]
@ooprathamm just double checking do you have pytng installed? Otherwise the tests will be skipped.

**Comment by hmacdope** [comment]
I verified on my own setup that warnings down from ~900 to ~80 so seems to work

**Review by hmacdope — APPROVED** [review]
LGTM!

**Review by hmacdope — CHANGES_REQUESTED** [review]
After some discussion @IAlibay  and myself have decided we want to eliminate all the warnings.
Try adding the same filter to more test fixture classes until the warning count gets to 0 for the "Stride of block" warnings.

**Comment by ooprathamm** [comment]
> I verified on my own setup that warnings down from ~900 to ~80 so seems to work

@hmacdope the difference in warnings seems to be from other optional dependencies in your system (for mine and as stated in #3986 ~600 to ~7 warnings)

is there list of optional dependencies that I need to install other than pytng

**Comment by hmacdope** [comment]
No just pytng should be fine! Try and get the last 7 gone and then we should be good to go.

**Comment by ooprathamm** [comment]
image
@hmacdope filtered all the warnings in test_tng
 also removed deprecation warning thrown for xdrlib in topology package

File: package/MDAnalysis/topology/TPRParser.py [hunk 2]
```diff
@@ -157,7 +157,11 @@
 __author__ = "Zhuyi Xue"
 __copyright__ = "GNU Public Licence, v2"
 
-import xdrlib
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import xdrlib
```
Comment (IAlibay) [inline package/MDAnalysis/topology/TPRParser.py]: Please keep this in, the xdrlib deprecation is not an internal warning and shouldn't be filtered out (it's something we'll fix soon on our own).

**Review by IAlibay — CHANGES_REQUESTED** [review]
Just the one thing, also please add yourself to AUTHORS and your name to the top of the changelog (a changelog entry is not required). We also would ask if you could introduce yourself on the mailing list if you haven't done so (please include your github handle).

File: package/MDAnalysis/topology/TPRParser.py [hunk 3]
```diff
@@ -157,7 +157,7 @@
 __author__ = "Zhuyi Xue"
 __copyright__ = "GNU Public Licence, v2"
 
-import xdrlib
+import warnings
```
Comment (IAlibay) [inline package/MDAnalysis/topology/TPRParser.py]: Sorry if I'm missing something obvious - is this new import being used anywhere?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Just the one thing from me.

File: package/MDAnalysis/topology/TPRParser.py (hunk 3 above)
Comment (ooprathamm) [reply package/MDAnalysis/topology/TPRParser.py]: that seems to be my mistake.

**Review by hmacdope — APPROVED** [review]
Looks good to me, I think you have already introduced yourself on the Mailing list, so good to go from me.

**Review by IAlibay — APPROVED** [review]
lgtm, thanks!

**Pull request merged**